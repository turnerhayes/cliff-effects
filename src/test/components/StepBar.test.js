import React from 'react';
import { shallow, mount } from 'enzyme';

import StepBar from '../../components/StepBar';

describe('<StepBar>', () => {
  const goToStep = jest.fn();
  const steps = [
    'firstStep',
    'secondStep',
    'thirdStep',
  ].map(
    (key) => {
      return {
        key,
        name: `${key} name`,
      };
    }
  );
  const defaultProps = {
    currentStepIndex: 1,
    goToStep:         goToStep,
    steps:            steps,
  };

  const buildWrapper = (props = {}, useMount = false) => {
    return (useMount ? mount : shallow)(
      (
        <StepBar
          { ...defaultProps }
          { ...props } />
      )
    );
  };

  afterEach(() => {
    goToStep.mockReset();
  });

  it('renders steps with active prop', () => {
    expect(buildWrapper({ currentStepIndex: 1 }).shallow().find('Step').at(0).prop('active')).toBe(true);
    expect(buildWrapper({ currentStepIndex: 2 }).shallow().find('Step').at(1).prop('active')).toBe(true);
    expect(buildWrapper({ currentStepIndex: 3 }).shallow().find('Step').at(2).prop('active')).toBe(true);
  });

  it('calls goToStep with expected index', () => {
    const wrapper = buildWrapper().shallow();

    wrapper.find('Step').at(0).simulate('click');
    expect(goToStep).toHaveBeenLastCalledWith(1);

    wrapper.find('Step').at(1).simulate('click');
    expect(goToStep).toHaveBeenLastCalledWith(2);

    wrapper.find('Step').at(2).simulate('click');
    expect(goToStep).toHaveBeenLastCalledWith(3);
  });

  it('matches snapshot', () => {
    expect(buildWrapper(undefined, true)).toMatchSnapshot();
  });
});
