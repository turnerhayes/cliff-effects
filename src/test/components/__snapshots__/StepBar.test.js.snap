// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepBar> matches snapshot 1`] = `
<StepBar
  currentStepIndex={1}
  goToStep={[Function]}
  steps={
    Array [
      Object {
        "key": "firstStep",
        "name": "firstStep name",
      },
      Object {
        "key": "secondStep",
        "name": "secondStep name",
      },
      Object {
        "key": "thirdStep",
        "name": "thirdStep name",
      },
    ]
  }
>
  <StepGroup
    className="six"
    items={
      Array [
        Object {
          "active": true,
          "key": 0,
          "onClick": [Function],
          "title": Object {
            "content": "firstStep name",
          },
        },
        Object {
          "active": false,
          "key": 1,
          "onClick": [Function],
          "title": Object {
            "content": "secondStep name",
          },
        },
        Object {
          "active": false,
          "key": 2,
          "onClick": [Function],
          "title": Object {
            "content": "thirdStep name",
          },
        },
      ]
    }
    ordered={true}
    size="mini"
  >
    <div
      className="ui mini ordered steps six"
    >
      <Step
        active={true}
        key="0"
        onClick={[Function]}
        title={
          Object {
            "content": "firstStep name",
          }
        }
      >
        <a
          className="active step"
          onClick={[Function]}
        >
          <StepContent
            title={
              Object {
                "content": "firstStep name",
              }
            }
          >
            <div
              className="content"
            >
              <StepTitle
                content="firstStep name"
              >
                <div
                  className="title"
                >
                  firstStep name
                </div>
              </StepTitle>
            </div>
          </StepContent>
        </a>
      </Step>
      <Step
        active={false}
        key="1"
        onClick={[Function]}
        title={
          Object {
            "content": "secondStep name",
          }
        }
      >
        <a
          className="step"
          onClick={[Function]}
        >
          <StepContent
            title={
              Object {
                "content": "secondStep name",
              }
            }
          >
            <div
              className="content"
            >
              <StepTitle
                content="secondStep name"
              >
                <div
                  className="title"
                >
                  secondStep name
                </div>
              </StepTitle>
            </div>
          </StepContent>
        </a>
      </Step>
      <Step
        active={false}
        key="2"
        onClick={[Function]}
        title={
          Object {
            "content": "thirdStep name",
          }
        }
      >
        <a
          className="step"
          onClick={[Function]}
        >
          <StepContent
            title={
              Object {
                "content": "thirdStep name",
              }
            }
          >
            <div
              className="content"
            >
              <StepTitle
                content="thirdStep name"
              >
                <div
                  className="title"
                >
                  thirdStep name
                </div>
              </StepTitle>
            </div>
          </StepContent>
        </a>
      </Step>
    </div>
  </StepGroup>
</StepBar>
`;
